### PR TITLE
Podcast show

### PR DESCRIPTION
**Pull request by janegreene**
Added episodes poros for view of podcast show page
used fixture files for now with 30 sec clips
Added PodcastFacade & service to break up responsibilities.

### CODE REVIEW

File: app/controllers/podcasts_controller.rb [hunk 1]
```diff
@@ -8,9 +8,14 @@ def new
     @podcast = Podcast.new
   end
 
+  def show
+    podcast = Podcast.find(params[:id])
+    @podcast_facade = PodcastFacade.new(podcast)
```
Comment (muydanny) [inline app/controllers/podcasts_controller.rb:13]: Great addition with the facade

File: app/facades/podcast_facade.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class PodcastFacade
+  attr_reader :podcast
+  def initialize(podcast_info)
```
Comment (muydanny) [inline app/facades/podcast_facade.rb:3]: podcast_Info a podcast object

File: app/facades/podcast_facade.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                'facebook' => @podcast[:facebook],
+                'patreon' =>  @podcast[:patreon],
+                'instagram' => @podcast[:instagram]
+              }
+    find = handles.find_all{ |k, v| v.nil? == false}
+    find.to_h
+  end
+
+  def episodes
+    formatted_episodes = Episode.make_episodes(@podcast.spotify_uri) #we want an array of episode name, description,  audio preview url
```
Comment (takeller) [inline app/facades/podcast_facade.rb:19]: Ditch comments?

File: app/facades/podcast_facade.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    test = formatted_episodes.map do |episode_hash|
```
Comment (takeller) [inline app/facades/podcast_facade.rb:20]: Remove test variable

File: app/models/episode.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  def self.make_episodes(spotify_uri)
+      podcast_service = PodcastService.new
+      episode_details = podcast_service.episodes(spotify_uri)
+      format_episode_details(episode_details)
+  end
+
+
+  private
+  def self.format_episode_details(episode_details)
+    episode_details[:episodes][:items][0..4].map do |details|
```
Comment (muydanny) [inline app/models/episode.rb:19]: I think first five is a great idea

File: app/models/podcast.rb [hunk 6]
```diff
@@ -3,6 +3,15 @@ class Podcast < ApplicationRecord
   has_many :podcast_tags, dependent: :destroy
   has_many :tags, through: :podcast_tags
   validates_presence_of :name
-  validates_presence_of :location 
+  validates_presence_of :location
   validates_presence_of :description
+
+  # def make_episodes
```
Comment (takeller) [inline app/models/podcast.rb:9]: Remove comments?

**Pull request merged**